app/src/main/java/com/example/tasky/agenda/presentation/components/AgendaItem.kt

```diff
@@ -0,0 +1,238 @@
+        modifier = modifier
+    )
+}
+
+@Composable
+fun TaskItem(
+    task: Agenda.Task,
+    onDoneClicked: (Boolean) -> Unit,
+    onMoreOptionsClicked: () -> Unit,
+    modifier: Modifier = Modifier,
+    isDone: Boolean = false,
+) {
+    AgendaItem(
+        title = task.title,
+        description = task.description,
+        date = task.date,
+        onDoneClicked = onDoneClicked,
+        onMoreOptionsClicked = onMoreOptionsClicked,
+        isTask = true,
+        isDone = isDone,
+        modifier = modifier
+    )
+}
+
+@Composable
+fun AgendaItem(
+    title: String,
+    description: String,
+    date: String,
+    modifier: Modifier = Modifier,
+    isTask: Boolean = false,
+    isDone: Boolean = false,
+    onDoneClicked: (Boolean) -> Unit,
+    onMoreOptionsClicked: () -> Unit
+) {
+    Card(
+        modifier = modifier
+            .fillMaxWidth()
+            .padding(start = 14.dp, end = 8.dp),
+        colors = CardDefaults.cardColors(
+            containerColor = Green,
+            contentColor = Color.White
+        )
+    ) {
+        Column(
+            modifier = Modifier.padding(
+                start = 15.dp,
+                top = 17.dp,
+                end = 15.dp,
+                bottom = 12.dp
+            ),
+        ) {
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+                horizontalArrangement = Arrangement.Center
+            ) {
+                if (isTask) {
+                    Column {
+                        Spacer(modifier = Modifier.size(5.dp))
+                        Icon(
+                            modifier = Modifier
+                                .size(20.dp)
+                                .toggleable(value = isDone) { isDone ->
+                                    onDoneClicked(isDone)
+                                },
+                            imageVector =
+                            if (isDone)
+                                Icons.Outlined.CheckCircle
+                            else
+                                Icons.Outlined.Circle,
+                            contentDescription = stringResource(
+                                R.string.agenda_item_check
+                            ),
+                        )
+                    }
+                }
+                Column(
+                    modifier = Modifier
+                        .padding(start = 12.dp)
+                        .weight(1f)
+                ) {
+                    Text(
+                        text = title,
+                        fontSize = 20.sp,
+                        fontWeight = FontWeight.Bold,
+                        textDecoration =
+                        if (isDone)
+                            TextDecoration.LineThrough
+                        else null
+                    )
+
+                    Spacer(modifier = Modifier.height(15.dp))
+
+                    Text(
+                        text = description,
```

Comment:
I'd add a MaxLines of 4 or similar to the description.
If a item has way to big of a description, then the item will be endless